force-app/nksAddresses/lwc/nksBostedAddress/nksBostedAddress.js

```diff
@@ -19,60 +19,40 @@ export default class NksBostedAddress extends LightningElement {
-    get iconName() {
-        return this.open ? 'utility:chevrondown' : 'utility:chevronright';
-    }
-
-    get hasRecords() {
-        return this.residentialAddresses.length > 0;
-    }
-
     get residentialAddresses() {
-        let addressesToReturn = [];
-        if (this._residentialAddresses.length > 0) {
-            this.showCopyButton = true;
-            this._residentialAddresses.forEach((element) => {
-                if (element.fullName) {
-                    addressesToReturn.push(element.fullName);
-                }
-                let addressLine = '';
-                if (element.address) {
-                    addressLine += element.address;
-                }
-                if (element.houseNumber) {
-                    addressLine += ' ' + element.houseNumber;
-                }
-                if (element.houseLetter) {
-                    addressLine += element.houseLetter;
-                }
-                addressesToReturn.push(addressLine);
-                let postInfo = '';
-                if (element.zipCode) {
-                    postInfo += element.zipCode;
-                }
-                if (element.city) {
-                    postInfo += ' ' + element.city;
-                }
-                addressesToReturn.push(postInfo);
-                let region = '';
-                if (element.region) {
-                    region += element.region;
-                }
-                if (element.countryCode) {
-                    region += ' ' + element.countryCode;
-                }
-                if (region !== '') {
-                    addressesToReturn.push(region);
-                } else {
-                    addressesToReturn.push('NORGE NO');
-                }
-            });
+        if (this._residentialAddresses.length === 0) {
+            return;
         }
-        return addressesToReturn.join('\n');
+
+        this.showCopyButton = true;
+        const addressesToReturn = this._residentialAddresses.map((element) => {
+            const type = element.type ? [element.type + ':'] : [];
+            const fullName = element.fullName ? [element.fullName] : [];
+        
+            const addressLine = [
+                element.address ? element.address : '',
+                element.houseNumber ? ' ' + element.houseNumber : '',
+                element.houseLetter ? ' ' + element.houseLetter : ''
+            ].join('').trim();
+        
+            const postInfo = [
+                element.zipCode ? element.zipCode : '',
+                element.city ? ' ' + element.city : ''
+            ].join('').trim();
+        
+            const region = [
+                element.region ? element.region : '',
+                element.countryCode ? ' ' + element.countryCode : ''
+            ].join('').trim();
+
+            return [...type, fullName, addressLine, postInfo, region || 'NORGE NO'].join('\n').trim();
+        });
+        return addressesToReturn.join('\n\n').trim();
```

Comment:
Detta e for advanced for mæ, æ må nesten se det i action før å forstår det. Too be continued ig.